src/pages/home/ReportScreen.js

```diff
@@ -168,8 +170,11 @@ function ReportScreen({
     const {addWorkspaceRoomOrChatPendingAction, addWorkspaceRoomOrChatErrors} = ReportUtils.getReportOfflinePendingActionAndErrors(report);
     const screenWrapperStyle = [styles.appContent, styles.flex1, {marginTop: viewportOffsetTop}];
 
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- need to re-filter the report actions when network status changes
+    const filteredReportActions = useMemo(() => _.filter(reportActions, (action) => ReportActionsUtils.shouldReportActionBeVisible(action)), [isOffline, reportActions]);
```

Comment:
I think iterating over report actions is going to be responsible for performance?

Reply:
@MonilBhavsar 

Yes, it may be. We can remove the selector function from the onyx connector [here](https://github.com/Expensify/App/blob/d3ea6eaca140c64df083f2f6ebae78c19437d9a2/src/pages/home/ReportScreen.js#L480) and call `getSortedReportActionsForDisplay` inside the `useMemo`.

What do you think?

cc @ArekChr